src/modules/cpu/host_tensor_augmentations.hpp

```diff
@@ -7765,817 +7725,865 @@ omp_set_dynamic(0);
+        Rpp32u heightLimit = roiPtr->xywhROI.roiHeight - 2;
+        Rpp32u widthLimit = roiPtr->xywhROI.roiWidth - 2;
+
+        Rpp16f *srcPtrChannel, *dstPtrChannel, *srcPtrImage, *dstPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        dstPtrImage = dstPtr + batchCount * dstDescPtr->strides.nStride;
+        srcPtrChannel = srcPtrImage + (roiPtr->xywhROI.xy.y * srcDescPtr->strides.hStride) + (roiPtr->xywhROI.xy.x * srcLayoutParams.bufferMultiplier);
+        dstPtrChannel = dstPtrImage;
+        Rpp32f srcLocationRow, srcLocationColumn, pixel;
+        Rpp32s srcLocationRowFloor, srcLocationColumnFloor;
+        Rpp32u alignedLength = dstImgSize[batchCount].width & ~3;
+        __m128 pWRatio = _mm_set1_ps(wRatio);
+        __m128 pDstLocInit = _mm_setr_ps(0, 1, 2, 3);
+        __m128 pWidthLimit = _mm_set1_ps((float)widthLimit);
+        __m128 pWeightParams[4], pBilinearCoeffs[4], pColFloor, pDstLoc;
+        __m128i pxColFloor;
+        Rpp32s srcLocCF[4] = {0};
+        Rpp32f weightParams[4], bilinearCoeffs[4];
+
+        // Resize with fused output-layout toggle (NHWC -> NCHW)
+        if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NHWC) && (dstDescPtr->layout == RpptLayout::NCHW))
+        {
+            Rpp16f *dstPtrRowR, *dstPtrRowG, *dstPtrRowB;
+            dstPtrRowR = dstPtrChannel;
+            dstPtrRowG = dstPtrRowR + dstDescPtr->strides.cStride;
+            dstPtrRowB = dstPtrRowG + dstDescPtr->strides.cStride;
+
+            for(int i = 0; i < dstImgSize[batchCount].height; i++)
+            {
+                Rpp16f *dstPtrTempR, *dstPtrTempG, *dstPtrTempB;
+                dstPtrTempR = dstPtrRowR;
+                dstPtrTempG = dstPtrRowG;
+                dstPtrTempB = dstPtrRowB;
+
+                Rpp16f *srcRowPtrsForInterp[2];
+                compute_resize_src_loc(i, hRatio, heightLimit, srcLocationRowFloor, &weightParams[0]);
+                srcRowPtrsForInterp[0] = srcPtrChannel + srcLocationRowFloor * srcDescPtr->strides.hStride;    // srcPtrTopRow for bilinear interpolation
+                srcRowPtrsForInterp[1]  = srcRowPtrsForInterp[0] + srcDescPtr->strides.hStride;                // srcPtrBottomRow for bilinear interpolation
+                pWeightParams[0] = _mm_set1_ps(weightParams[0]);
+                pWeightParams[1]  = _mm_set1_ps(weightParams[1]);
+                pDstLoc = pDstLocInit;
+
+                int vectorLoopCount = 0;
+                for (; vectorLoopCount < alignedLength; vectorLoopCount += 4)
+                {
+                    __m128 pRow[12], pPixels[3];
+
+                    compute_resize_src_loc_sse(pDstLoc, pWRatio, pWidthLimit, srcLocCF, &pWeightParams[2], true);
+                    compute_bilinear_coefficients_sse(pWeightParams, pBilinearCoeffs);
+
+                    rpp_simd_load(rpp_bilinear_load4_f16pkd3_to_f32pln3, srcRowPtrsForInterp, srcLocCF, pRow);
```

Comment:
Just for F16, lets actually get rid of any additional functions like rpp_bilinear_load4_f16pkd3_to_f32pln3() or the store func. Lets use the same for loops here in this file along with same F32 calls to rpp_bilinear_load4_f32pkd3_to_f32pln3() and the store func for f32. Like - https://github.com/r-abishek/rpp/blob/915707da5bc05ace925e979849b63cfecafe84cd/src/modules/cpu/host_tensor_augmentations.hpp#L6136-L6152

Reply:
This is since the f16/f32 type cast is quite suboptimal and we'll be changing the whole mechanism for all functions in the near future.

Reply:
rpp_bilinear_load4_f32pkd3_to_f32pln3() will actually load the pixels based on the location and store it in vectors.
rpp_bilinear_load4_f16pkd3_to_f32pln3() would load the pixels based on location value then cast it to float and store in vectors.
The required source pixels are not loaded from contiguous memory but is influenced by the location factor.
So for loops followed by rpp_bilinear_load4_f32pkd3_to_f32pln3() would not work.